### PR TITLE
Vibrant window for dark theme

### PR DESCRIPTION
**Pull request by oskarkrawczyk**
screen shot 2016-10-29 at 18 14 21

If something's not working run:

```
npm run rebuild
```

### CODE REVIEW

File: browser.js [hunk 1]
```diff
@@ -185,7 +185,7 @@ document.addEventListener('DOMContentLoaded', () => {
 	// prevent flash of white on startup when in dark mode
 	// TODO: find a CSS only solution
 	if (config.get('darkMode')) {
-		document.documentElement.style.backgroundColor = '#192633';
+		document.documentElement.style.backgroundColor = 'rgba(0,0,0,0)';
```
Comment (sindresorhus) [inline browser.js]: `rgba(0,0,0,0)` => `rgba(0, 0, 0, 0)`

File: package.json [hunk 2]
```diff
@@ -34,12 +34,14 @@
...
-    "electron-dl": "^1.0.0"
+    "electron-dl": "^1.0.0",
+    "electron-vibrancy": "^0.1.0"
   },
   "devDependencies": {
     "electron": "^1.4.0",
     "electron-packager": "^8.0.0",
-    "xo": "*"
+    "xo": "*",
+    "electron-vibrancy": "^0.1.0"
```
Comment (sindresorhus) [inline package.json]: It shouldn't be here.

**Comment by oskarkrawczyk** [comment]
> Why only for the dark theme?

No reason, looks good on the dark one.

> Most apps using vibrancy only does it for the sidebar. See the Notes.app for example.

Most do, indeed. I like vibrancy so I use it for the whole window.

> The vibrancy effect is very buggy. It keeps glitching when I hover over different elements.

Interesting, I'm not getting this.

Let me know if you're willing to pull this in if I fix the mentioned things. Otherwise go ahead and close.

**Comment by sindresorhus** [comment]
> Interesting, I'm not getting this.

Hmm, seems to have gone away. Maybe because I bumped Electron.

> Let me know if you're willing to pull this in if I fix the mentioned things. Otherwise go ahead and close.

Initial impression, I like it. Gonna run for a little while and see how I feel about it then.

-

I think the navbar should be made vibrant too.

**Comment by sindresorhus** [comment]
Would appreciate it if people looking here could share their opinion on this.

## Before

screen shot 2016-11-04 at 23 54 48



## After
screen shot 2016-11-04 at 23 54 19

**Comment by oskarkrawczyk** [comment]
> Hmm, seems to have gone away. Maybe because I bumped Electron.

I've noticed there's some artifacting when playing inline video – most likely due the fact that it's hardware accelerated.

> I think the navbar should be made vibrant too.

Actually, the nav-bar is semi-transparent, but the interface is built in a way that scrollable content doesn't go under it. And even if it did, I've tried turning on Chromium experimental features and adding `backdrop-filter:blur` but got a lot of artifacting (again, most likely because it's triggering GPU to do the drawing)  

> Initial impression, I like it. Gonna run for a little while and see how I feel about it then.

Sure. I've been running it for some time before creating a PR, just to see if the full-window vibrancy won't get annoying but it didn't. But maybe it's because I'm a sucker for all that blurs!

**Comment by oskarkrawczyk** [comment]
Also, I think it should be possible to have Vibrancy an option, just like dark mode.

**Comment by sindresorhus** [comment]
I tweeted about getting some opinions on this: https://twitter.com/sindresorhus/status/794584600664150017

**Comment by sindresorhus** [comment]
> Also, I think it should be possible to have Vibrancy an option, just like dark mode.

I prefer the least amount of option. So not gonna do an option unless vibrancy is very unpopular or split opinions about it.

**Comment by ashtonwar** [comment]
Does it still look pretty when the background isn't a desktop wallpaper? ie with another window / application behind it.

**Comment by oskarkrawczyk** [comment]
@ashtonwar
screen shot 2016-11-04 at 18 23 56

**Comment by kimroen** [comment]
I enjoy this look - I would use it.

**Comment by terkelg** [comment]
ike itl I – the application feels more lightweight, but it's really just a matter of taste.

**Comment by ivancuric** [comment]
Maybe just for the sidebar... Not sure. I'm not a fan of transparent interfaces. I feel the same way about transparent terminal windows.

**Comment by ZakariaRidouh** [comment]
@ivancuric I guess there should be a simple option giving the choice?

**Comment by danhp** [comment]
I have a preference for keeping the vibrancy contained to the sidebar so that it matches the rest of the system. Also https://github.com/electron/electron/pull/7898 has landed and is released as of `v1.4.7` so I'd use that instead of the module.

**Pull request closed without merging**